client/dashboard/components/switcher/switcher-content.tsx

```diff
@@ -48,31 +50,53 @@ export default function SwitcherContent< T >( {
 	resetScroll?: boolean;
 	onClose: () => void;
 	onItemClick?: () => void;
+	Filter?: JSX.Element;
+	filterField?: Field< T >;
 } > ) {
 	const fields = useMemo( () => {
-		return searchableFields.map( ( searchableField ) => ( {
+		const allFields = searchableFields.map( ( searchableField ) => ( {
 			...searchableField,
 			enableGlobalSearch: true,
 		} ) );
-	}, [ searchableFields ] );
+
+		if ( filterField ) {
+			allFields.push( {
+				...filterField,
+				enableGlobalSearch: false,
+			} );
+		}
+
+		return allFields;
+	}, [ searchableFields, filterField ] );
 
 	if ( ! items ) {
 		return __( 'Loading…' );
 	}
 
 	const { data: filteredData } = filterSortAndPaginate( items, view, fields );
 
+	const Search = (
+		<SearchControl
+			className={ searchClassName }
+			label={ __( 'Search' ) }
+			value={ view.search }
+			onChange={ ( value ) => onChangeView( { ...view, search: value } ) }
+			size="compact"
+			__nextHasNoMarginBottom
+		/>
+	);
+
 	return (
 		<NavigableMenu style={ { width } }>
 			<MenuGroup>
-				<SearchControl
-					className={ searchClassName }
-					label={ __( 'Search' ) }
-					value={ view.search }
-					onChange={ ( value ) => onChangeView( { ...view, search: value } ) }
-					size="compact"
-					__nextHasNoMarginBottom
-				/>
+				{ Filter ? (
+					<HStack justify="flex-start" alignment="center">
+						{ Search }
```

Comment:
Nit: I think `Search` and `Filter` should be just `search` and `filter`, because they are React elements, not React components.